src/components/starlight/Sidebar.astro

```diff
@@ -32,8 +32,7 @@ async function remap(i: SidebarEntry) {
 sidebar = await Promise.all(sidebar.map(remap));
 ---
 
-<SidebarSublist sublist={sidebar} />
-
+<Default {...Astro.props} sidebar={sidebar} />
 <Sponsors />
 
 <div class="md:sl-hidden">
```

Comment:
I think it’s going to be a bit trickier than this because the sidebar already includes the mobile menu footer re-rendered below here, so on mobile this change results in:

<img width="518" alt="Bottom of a mobile menu showing a footer with colour scheme and language options being rendered twice" src="https://github.com/user-attachments/assets/6a33a8b7-347d-47b3-a44e-ac54fc69a319">

Might need to remove the footer from this file? But not quite sure yet what to do with the sponsors 🤔

Reply:
Yeah. Maybe we need to refactor Starlight's Sidebar component? So that it's easier for people to override it.

I tried to migrate the sl-sidebar-state-persist logic to the SideBarSublist component and it still seems to be working!

![image](https://github.com/user-attachments/assets/361b24dc-9e9f-4aa2-a30a-5ef665368e87)

Reply:
The issue there is that `<SidebarSublist>` is rendered recursively, so having the script there means you will get it rendered for every sidebar group, which is not what we want.

The easiest quick fix might be to:
- Remove the extra mobile footer in our override
- Hide the `<Sponsors/>` on mobile

Then later we can figure out if sponsors on mobile is important and how to bring them back.

Reply:
I've temporarily hidden the `<Sponsors/>` component on mobile.